### ISSUE TITLE
options: except *route*

### ISSUE THREAD

**Issue by fbender** [franciscop/pagex#2]
Similar to what was being proposed here: https://news.ycombinator.com/item?id=12303463

For me, the negate parameter is the wrong way around, so I'd rather have the default be `route_when_match == true` with `route_when_match == false` being equivalent to `negate == true`. But either way it is much more readable with an `options` parameter (which you can easily implement in a backwards-compatible way). With that, I'd like to see a more flexible solution for the negate use case:

```
pagex(path [, options ], callback);
options = {
    'negate': false,
    'except': []
}
```

The `except` option will take a list of `path`s that, when the route matches against any of these, the `callback` will not execute. The case `negate == false` will then equate to `except == [path]` and (internally) `path == /.*/`. 

Furthermore, you may want to react to a route not or _no longer_ matching (e.g. locking a user session as soon as s/he leaves a "subdir" resp. accesses a different "subdir"), which is especially useful in the `negate` case. So you may either want to always execute the `callback` and pass some `matches?` parameter to it or (the probably more efficient and also backwards-compatible route) have another callback that is executed in the non-matching case: 

```
pagex(path [, options ], callback, unmatched_callback);
```

**Comment by franciscop** [comment]
I totally agree with the options as an object instead of the boolean, but I don't really like the except as it seems overkill. Also I want to allow for multiple callbacks, which would also not be compatible with the `unmatched_callback`

**Comment by fbender** [comment]
You should keep `negate` as-is (with the `options` parameter) for the general case. The idea with `except` is that you can easily specify more than one route where a callback will not fire. Plus, the `except` can narrow the cases where a callback will fire even for the normal, non-negate use case (i. e. providing a blacklist for a handler). 

One thing that needs to be considered for `except` is whether its meaning is reversed when it's used in conjunction with `negate == true`, i. e. "execute handler when `path` does _not_ match except when it matches against one of the `exclude`s". This is a bit more trivial to implement than "execute handler when `path` does _not_ match and also when it matches against one of the `exclude`s". If you prefer the latter behaviour, then `exclude` should probably be renamed (e. g. `never_match`?) and disallow specifying it alongside `negate`. 

The `unmatch_callback` could be specified in the `options` object instead, so that'll be compatible with your plan.
